controller/lib/debug/vars.cpp

```diff
@@ -0,0 +1,22 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "vars.h"
+#include "debug.h"
+#include <string.h>
+
+// static member variables
+UntypedDebugVar *DebugVarRegistry::all_vars[100] = {nullptr};
+int DebugVarRegistry::var_count = 0;
```

Comment:
If you like these can be `inline` variables, then you don't need the cpp file.  Up to you.